src/Scientist/Internals/GCHelper.cs

```diff
@@ -0,0 +1,132 @@
+#endif
+
+        private static long AllocationQuantum { get; } = CalculateAllocationQuantumSize();
+
+        private static readonly Func<long> GetAllocatedBytesForCurrentThreadDelegate = GetAllocatedBytesForCurrentThread();
+
+        /// <summary>
+        /// returns total allocated bytes (not per operation)
+        /// </summary>
+        /// <param name="excludeAllocationQuantumSideEffects">Allocation quantum can affecting some of our nano-benchmarks in non-deterministic way.
+        /// when this parameter is set to true and the number of all allocated bytes is less or equal AQ, we ignore AQ and put 0 to the results</param>
+        /// <returns></returns>
+        public static long GetAllocatedBytes()
+        {
+            var allocation = GetAllocatedBytesByRuntime();
+
+            if (IsNetCore)
+            {
+                return allocation;
+            }
+
+            return allocation <= AllocationQuantum ? 0L : allocation;
+        }
+
+        private static long GetAllocatedBytesByRuntime()
+        {
+            // Monitoring is not available in Mono, see http://stackoverflow.com/questions/40234948/how-to-get-the-number-of-allocated-bytes-
+            if (IsMono)
+            {
+                return 0;
+            }
+            //GC.Collect();
+
+#if (NET451)
+            // "This instance Int64 property returns the number of bytes that have been allocated by a specific 
+            // AppDomain. The number is accurate as of the last garbage collection." - CLR via C#
+            // so we enforce GC.Collect here just to make sure we get accurate results
+            return AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
+#elif (NETSTANDARD2_0)
+            // it can be a .NET app consuming our .NET Standard package
+            if (IsFullFramework) 
+            {
+                GC.Collect();
+                return AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
+            }
+            // https://apisof.net/catalog/System.GC.GetAllocatedBytesForCurrentThread() is not part of the .NET Standard, so we use reflection to call it..
+            return GetAllocatedBytesForCurrentThreadDelegate.Invoke();
+#elif (NETCOREAPP2_1)
+            // but CoreRT does not support the reflection yet, so only because of that we have to target .NET Core 2.1
+            // to be able to call this method without reflection and get MemoryDiagnoser support for CoreRT ;)
+            return GC.GetAllocatedBytesForCurrentThread();
+#else
+            return 0;
+#endif
+        }
+
+        private static Func<long> GetAllocatedBytesForCurrentThread()
+        {
+            // for some versions of .NET Core this method is internal, 
+            // for some public and for others public and exposed ;)
+#if (NETSTANDARD2_0 || NETCOREAPP2_1)
+            var method = typeof(GC)
```

Comment:
Can we leverage `MethodInfo.CreateDelegate(typeof(Func<long>))` to create a delegate from the `MethodInfo`? I'm unsure if there are any limitations based on .NET Framework vs .NET Core.

Reply:
The issue here is that the .Net Core team soft launched the new API feature as internal before finally making it public sometime later. This code is looking for the method in either form (hence the public/nonpublic binding flags). You could do it the other way, but you'd drop support prior to Net Core 2.1.

Reply:
Not seen that method before, I've swapped out the implementation for CreateDelegate.